src/components/Tools/ColorPicker/utils.ts

```diff
@@ -0,0 +1,77 @@
+import type { Color } from './types'
+
+// Convert hex to RGB and HSL
+export function hexToFormats(hex: string): Color {
+  const r = Number.parseInt(hex.slice(1, 3), 16)
+  const g = Number.parseInt(hex.slice(3, 5), 16)
+  const b = Number.parseInt(hex.slice(5, 7), 16)
+
+  // RGB format
+  const rgb = `rgb(${r}, ${g}, ${b})`
+
+  // HSL calculation
+  const rn = r / 255
+  const gn = g / 255
+  const bn = b / 255
+  const max = Math.max(rn, gn, bn)
+  const min = Math.min(rn, gn, bn)
+  let h = 0
+  let s = 0
+  const l = (max + min) / 2
+
+  if (max !== min) {
+    const d = max - min
+    s = l > 0.5 ? d / (2 - max - min) : d / (max + min)
+
+    if (max === rn)
+      h = (gn - bn) / d + (gn < bn ? 6 : 0)
+    else if (max === gn)
+      h = (bn - rn) / d + 2
+    else h = (rn - gn) / d + 4
+
+    h /= 6
+  }
+
+  const hsl = `hsl(${Math.round(h * 360)}, ${Math.round(
+    s * 100,
+  )}%, ${Math.round(l * 100)}%)`
+  const time = new Date().toLocaleTimeString([], {
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  })
+
+  return { hex, rgb, hsl, time }
+}
+
+// Copy to clipboard with visual feedback
+export async function copyToClipboard(text: string) {
+  await navigator.clipboard.writeText(text)
+  // Simple feedback without external alert system
+  const btn = document.activeElement as HTMLElement
+  if (btn) {
+    btn.classList.add('scale-75')
```

Comment:
This is probably introducing a small bug when you click on the picked color box. There's also an interesting behavior: even when the extension frame is closed, the color picker still shows some color hints, please check the video

https://github.com/user-attachments/assets/841a8eb5-66d3-40f4-aa30-3b6509dd44e5

Reply:
Yes, I'm using fedora